sources/core/objects/Gun.cpp

```diff
@@ -79,6 +79,7 @@ void Gun::ShootImpl(VDir target, const QString& sound,
             auto new_tile = tile->GetNeighbour(shooting_direction);
```

Comment:
I think it was a mistake to create a projectile on the nearby tile instead of the human tile.
It have seemed a good idea, but actually it causes this issue - and I cannot think up a good way to solve it without changing the place of the creation of projectiles.

It is needed to create the projectile on `tile` instead of `new_tile` and introduce a new variable into `Projectile` - `bool harmless_;`.
On a projectile creation it should be `true`, but after the first movement it will be set as `false`.
If `harmless` equal to `true` then `CheckHumanTile` should not be called, so the shooter will not be hit by the projectile.